programs/sunrise-stake/src/lib.rs

```diff
@@ -121,7 +143,7 @@ pub mod sunrise_stake {
         burn(
             lamports,
             &ctx.accounts.gsol_mint.to_account_info(),
-            &ctx.accounts.gsol_token_account_authority,
+            &ctx.accounts.gsol_mint_authority.to_account_info(),
```

Comment:
looks wrong to me? The authority for a burn operation should be the token account owner whose tokens are being burned.

Reply:
My bad. Silly mistake